File: ip-messaging/channels/join-a-channel/join-a-channel.m
// Where "client" is an authenticated client obtained with
// [TwilioChatClient chatClientWithToken:properties:delegate:]
// and the callback for chatClient:synchronizationStatusChanged: has been called with TCHClientSynchronizationStatusChannelsListCompleted
TCHChannel *channel = [[self.client channelsList] channelWithUniqueName:@"general"];
if (channel) {
    [channel joinWithCompletion:^(TCHResult *result) {
        if ([result isSuccessful]) {
            NSLog(@"Channel joined.");
        } else {
            NSLog(@"Channel NOT joined.");
        }
    }];
}

File: ip-messaging/channels/join-a-channel/join-a-channel.swift
// Where "client" is an authenticated client obtained with
// TwilioChatClient(token:, properties:, delegate:)
// and the callback for chatClient(client:, synchronizationStatusChanged:) has been called with .completed
let channel = client.channelsList().channel(withSidOrUniqueName:"general")
if let channel = channel {
    channel.join(completion: { channelResult in
        if channelResult.isSuccessful() {
            print("Channel joined.")
        } else {
            print("Channel NOT joined.")
        }
    }
}
